services/integrationtest.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+USAGE="$0 <test-executable> [args...]"
+EXE=${1:?$USAGE}
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+PARENT_PID=$$
+EXIT_STATUS_LOCATION=/dev/shm/integration_test_rc
+echo 1 >${EXIT_STATUS_LOCATION}
+
+function kill_all() {
+    # kill the process tree of the PARENT_PID
+    kill -9 -${PARENT_PID} &> /dev/null
+}
+
+function list_descendants () {
+  local children=$(ps -o pid= --ppid "$1")
```

Comment:
This doesn't work everywhere unfortunately (BSD systems at least).
I got it working for me by changing it to:
```
local children=$(pgrep -P "$1")
```

All tests pass although test runs finish with `{service}: write: resource vanished (Broken pipe)`

Reply:
Changed the local_children bit to be compatible, and also added a `pkill services`. Do you still get "Broken pipe", and do you get it for all services consistently?